src/Maybe.js

```diff
@@ -20,7 +20,7 @@ util.extend(Nothing, Maybe);
 Nothing.prototype.isNothing = true;
 Nothing.prototype.isJust = false;
 
-var _nothing = new Nothing();
+var _nothing = Object.freeze(new Nothing());
```

Comment:
Which browsers does this project target? Is `Object.freeze` available in all of them?

It seems simpler to me to update the implementation of `Maybe.Nothing`:

```javascript
Maybe.Nothing = function() {
  return new Nothing();
};
```